cmake/modules/RootNewMacros.cmake

```diff
@@ -827,13 +827,70 @@ function(ROOT_INSTALL_HEADERS)
 endfunction()
 
 #---------------------------------------------------------------------------------------------------
-#---ROOT_STANDARD_LIBRARY_PACKAGE(libname DEPENDENCIES lib1 lib2)
+#---ROOT_STANDARD_LIBRARY_PACKAGE(libname
+#                                 [NO_INSTALL_HEADERS]         : don't install headers for this package
+#                                 [STAGE1]                     : use rootcling_stage1 for generating
+#                                 HEADER header1 header2       : if not specified, globbing for *.h is used)
+#                                 SOURCES source1 source2      : if not specified, globbing for *.cxx is used)
+#                                 LIBRARIES lib1 lib2          : linking flags such as dl, readline
+#                                 DEPENDENCIES lib1 lib2       : dependencies such as Core, MathCore
+#                                 BUILTINS builtin1 builtin2   : builtins like AFTERIMAGE
+#                                 LINKDEF LinkDef.h LinkDef2.h : linkdef files
+#                                 DICTIONARY_OPTIONS option    : options passed to rootcling
+#                                 INSTALL_OPTIONS option       : options passed to install headers
+#                                )
 #---------------------------------------------------------------------------------------------------
 function(ROOT_STANDARD_LIBRARY_PACKAGE libname)
-  CMAKE_PARSE_ARGUMENTS(ARG "" "" "DEPENDENCIES;DICTIONARY_OPTIONS" ${ARGN})
-  ROOT_GENERATE_DICTIONARY(G__${libname} *.h MODULE ${libname} LINKDEF LinkDef.h OPTIONS ${ARG_DICTIONARY_OPTIONS})
-  ROOT_LINKER_LIBRARY(${libname} *.cxx G__${libname}.cxx DEPENDENCIES ${ARG_DEPENDENCIES})
-  ROOT_INSTALL_HEADERS()
+  set(options OPTIONAL NO_INSTALL_HEADERS STAGE1 HEADER_ONLY OBJECT_LIBRARY)
+  set(oneValueArgs)
+  set(multiValueArgs DEPENDENCIES HEADERS SOURCES BUILTINS LIBRARIES DICTIONARY_OPTIONS LINKDEF INSTALL_OPTIONS)
+  CMAKE_PARSE_ARGUMENTS(ARG "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+  # Check if we have any unparsed arguments
+  if(ARG_UNPARSED_ARGUMENTS)
+    message(AUTHOR_WARNING "Unparsed arguments for ROOT_STANDARD_LIBRARY_PACKAGE: ${ARG_UNPARSED_ARGUMENTS}")
+  endif()
+
+  if (NOT ARG_HEADERS)
+    set(ARG_HEADERS "*.h")
+  endif()
+  if (NOT ARG_SOURCES AND NOT ARG_HEADER_ONLY)
+    set(ARG_SOURCES "*.cxx")
```

Comment:
Yeah, but I didn't want to change ROOT_STANDARD_LIBRARY_PACKAGE behavior (right now it behaves exactly the same way as before as long as you don't use the new arguments). Maybe we should indeed use a new function. Let's discuss this on monday.